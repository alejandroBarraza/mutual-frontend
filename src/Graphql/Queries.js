import { gql } from '@apollo/client';

//videos queries(always get last created)
export const GETVIDEOS = gql`
    query Getvideo {
        videos(limit: 1, sort: "created_at:desc") {
            created_at
            titulo
            descripcion
            links
        }
    }
`;

//statistic query(always get last 4 created)
export const GETSTATISTICS = gql`
    query GetStatistic {
        statistics(limit: 4, sort: "created_at:asc") {
            titulo
            descripcion
            created_at
        }
    }
`;

// get download content given limit and start =="offset" as arguments
export const GETDOWNLOADS_LIMIT_START = gql`
    query GetDownloads($start: Int!, $limit: Int!) {
        downloads(start: $start, limit: $limit) {
            id
            published_at
            titulo
            imagenes {
                name
                url
            }
        }
    }
`;

export const GETDOWNLOADS_SORTED_DESC = gql`
    query GetDownloads {
        downloads(sort: "created_at:desc") {
            id
            titulo
            descripcion
            archivo {
                url
            }
        }
    }
`;

export const GETNEWS_START_LIMIT = gql`
    query getNews($limit: Int!, $start: Int!) {
        news(limit: $limit, start: $start, sort: "created_at:desc") {
            id
            titulo
            descripcion
            tiempoLectura
            imagen {
                url
            }
        }
    }
`;

// get the footer elements
export const GET_ALL_FOOTER = gql`
    query getFooter {
        footer {
            selloSeguridad
            descripcion
            instagram
            facebook
            twitter
            youtube
        }
    }
`;
