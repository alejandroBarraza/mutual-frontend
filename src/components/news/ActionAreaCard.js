import * as React from 'react';
import Card from '@mui/material/Card';
import CardMedia from '@mui/material/CardMedia';
import { CardActionArea } from '@mui/material';
import { useParams } from 'react-router';

export const ActionAreaCard = ({ download }) => {
   const handleDownload = () => {
      window.open(
         download.archivo.url,
         '_blank',
         'noopener,resizable,scrollbars'
      )
   }
   return (
      <Card sx={{ width: '100%' }} onClick={handleDownload}>
         <CardActionArea>
            <CardMedia
               component="img"
               height="140"
               image={`${download.imagen.url}`}
               alt="hola"
            />
         </CardActionArea>
      </Card>
   );
}