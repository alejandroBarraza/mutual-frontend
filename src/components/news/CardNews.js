import React from 'react';
import PropTypes from 'prop-types';

import { Link } from 'react-router-dom';

import { Card, Divider, CardContent, CardMedia, Typography } from '@mui/material';
import { Box } from '@mui/system';

import MenuBookRoundedIcon from '@mui/icons-material/MenuBookRounded';
import IosShareRoundedIcon from '@mui/icons-material/IosShareRounded';

import { styles } from './StyleCardNews';

export const CardNews = ({ descriptionInfo, data, maxWidth }) => {
    const { id, descripcion, imagen, titulo, tiempoLectura } = data;
    return (
        data && (
            <Link style={{ textDecoration: 'none' }} to={`../noticias/${id}`}>
                <Card sx={{ ...styles.card, maxWidth }}>
                    <CardMedia
                        component='img'
                        height='140rem'
                        image={imagen.url}
                        alt='testing pic'
                        sx={styles.media}
                    />
                    <CardContent>
                        <Typography gutterBottom variant='h5' component='div'>
                            {titulo}
                        </Typography>
                        {descriptionInfo && (
                            <Typography variant='body2' color='text.secondary' sx={{ height: '7rem', overflow: 'hidden' }}>
                                {descripcion}
                            </Typography>
                        )}
                    </CardContent>

                    <Divider sx={{ marginTop: 'auto' }} />
                    <Box sx={styles.readInfo}>
                        <Typography variant='subtitle2'>{`${tiempoLectura}min`} </Typography>
                        <MenuBookRoundedIcon fontSize='small' color='action' />
                        <IosShareRoundedIcon
                            fontSize='small'
                            color='action'
                            sx={{ marginLeft: ' auto' }}
                        />
                    </Box>
                </Card>
            </Link>
        )
    );
};

CardNews.propTypes = {
    maxWidth: PropTypes.string,
    columnNumber: PropTypes.object,
};
CardNews.defaultProps = {
    maxWidth: '345',
    columnNumber: {
        md: 4,
        sm: 6,
        xs: 12,
    },
};
