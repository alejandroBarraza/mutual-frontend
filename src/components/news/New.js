import React from 'react';

export const New = () => {
    return (
        <div>
            <h1>hola</h1>
        </div>
    );
};
