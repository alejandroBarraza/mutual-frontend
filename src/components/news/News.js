import React from 'react';

export const News = () => {
    return (
        <div>
            <h1>Testing new component for navegation</h1>
        </div>
    );
};
