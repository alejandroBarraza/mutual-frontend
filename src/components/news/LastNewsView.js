import React from 'react';
import { useQuery } from '@apollo/client';

import { Grid, Typography } from '@mui/material';

import { Box } from '@mui/system';

import { GETLASTNEWS } from '../../Graphql/Queries';
import { ErrorUI } from '../utils/ErrorUI';
import { Loading } from '../utils/Loading';
import { styles } from './StyleCardNewsView';
import { LastNews } from './LastNews';

export const LastNewsView = () => {

    const { data, loading, error } = useQuery(GETLASTNEWS, {
        variables: { start: 0, limit: 3 },
    });

    if (loading) return <Loading />;
    if (error) return <ErrorUI error={error.message} />;

    return (
        <div className='container'>
            <Box textAling='center' sx={styles.titleContainer}>
                <Typography variant='h4'>Ultimas Noticias</Typography>
            </Box>

            <Grid container spacing={6} alignItems='stretch'>
                {data.news.map((newData) => (
                    <Grid key={newData.id} item xs={12} sm={6} md={4} sx={{ align: 'center' }}>
                        <LastNews
                            descriptionInfo
                            key={newData.id}
                            data={newData}
                            maxWidth={'345'}
                        />
                    </Grid>
                ))}
            </Grid>
        </div>
    );
};

