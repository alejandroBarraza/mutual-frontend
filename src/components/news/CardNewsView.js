import React from 'react';
import { useQuery } from '@apollo/client';

import { Grid, Button, Typography } from '@mui/material';

import { Box } from '@mui/system';

import { GETNEWS_START_LIMIT } from '../../Graphql/Queries';
import { ErrorUI } from '../utils/ErrorUI';
import { Loading } from '../utils/Loading';
import { styles } from './StyleCardNewsView';
import { CardNews } from './CardNews';

export const CardNewsView = () => {
    const handleFetchMore = () => {
        fetchMore({
            variables: {
                start: data.news.length,
                limit: 3,
            },
            updateQuery: (prevValue, { fetchMoreResult }) => {
                if (!fetchMoreResult) {
                    return prevValue;
                }

                return {
                    news: [...prevValue.news, ...fetchMoreResult.news],
                };
            },
        });
    };

    const { data, loading, error, fetchMore } = useQuery(GETNEWS_START_LIMIT, {
        variables: { start: 0, limit: 3 },
    });

    if (loading) return <Loading />;
    if (error) return <ErrorUI error={error.message} />;

    return (
        <div style={{
            background: '#e5e5e583',
            paddingTop: "2rem",
            paddingBottom: "2rem",
        }}>
            <div className='container'>
                <Box textAling='center' sx={styles.titleContainer}>
                    <Typography variant='h3'>Ultimas Noticias</Typography>
                    <Typography variant='h5' align='center'>
                        todo acerca de las ultimas noticas y eventos Aun te espero
                    </Typography>
                </Box>

                <Grid container spacing={6} alignItems='stretch'>
                    {data.news.map((newData) => (
                        <Grid key={newData.id} item xs={12} sm={6} md={4} sx={{ align: 'center' }}>
                            <CardNews
                                descriptionInfo
                                key={newData.id}
                                data={newData}
                                maxWidth={'345'}
                            />
                        </Grid>
                    ))}
                </Grid>

                <Box textAlign='center' m={8}>
                    <Button variant='contained' color='success' onClick={handleFetchMore}>
                        Ver mas
                    </Button>
                </Box>
            </div>
        </div>

    );
};

// add after
// import debounce from 'lodash.debounce';
// const debouncedChangeHandler = useCallback(debounce(handleFetchMore, 1000), []);
