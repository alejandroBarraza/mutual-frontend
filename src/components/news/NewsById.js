import { useQuery } from '@apollo/client';
import React, { useEffect, useState } from 'react';
import { useParams } from 'react-router-dom';
import { GETNEWSBYID, GETNEWS_START_LIMIT } from '../../Graphql/Queries';
import showdown from 'showdown';
import { Markup } from 'interweave';
import { Box, Container, Grid } from '@mui/material';
import Typography from '@mui/material/Typography';
import Skeleton from '@mui/material/Skeleton';
import useMediaQuery from '@mui/material/useMediaQuery';
import { styles } from './StyleCardNewsView';
import { CardNews } from './CardNews';
import { PriorityDownloads } from '../download/PriorityDownloads';
import { Loading } from '../utils/Loading';

const converter = new showdown.Converter();

export const NewsById = () => {
    const { id } = useParams();
    /**
     * Database query.
     * @param loading indicates if data are still being downloaded.
     * @param error indicates if any error occurred.
     * @param data the data returned by the database.
     */
    const {
        loading: loadingData,
        error: errorData,
        data,
    } = useQuery(GETNEWSBYID, {
        variables: { new: parseInt(id) },
    });
    const {
        data: dataNews,
        loading,
        error,
    } = useQuery(GETNEWS_START_LIMIT, {
        variables: { start: 0, limit: 3 },
    });
    const [title, setTitle] = useState('');
    const [presentationImage, setPresentationImage] = useState('#');
    const [data_procesed, setData_procesed] = useState([]);
    const matches = useMediaQuery('(min-width:60rem)');

    useEffect(() => {
        setPresentationImage('#');
        if (data) {
            // // console.log(data);
            if (data.news != null && data.news.length !== 0) {
                let data_procesing = [];
                let html = converter.makeHtml(data.news[0].cuerpo);

                if (data.news[0].titulo != null) {
                    setTitle(data.news[0].titulo);
                }
                if (data.news[0].imagenDePresentacion != null) {
                    setPresentationImage(data.news[0].imagenDePresentacion.url);
                }

                html = html.split('<p>').filter((text) => {
                    return text !== '';
                });
                html.forEach((element) => {
                    element = element.split('</p>').filter((text) => {
                        return text !== '' && text !== '\n';
                    });
                    data_procesing = [...data_procesing, ...element];
                });

                setData_procesed([...html]);
            }
        }
    }, [data]);

    if (loading || loadingData) {
        return <Loading />;
    }
    if (errorData || error) {
        return (
            <Box
                sx={{
                    minHeight: '72vh',
                    width: '100%',
                    position: 'relative',
                    pb: '2rem',
                    display: 'flex',
                    justifyContent: 'center',
                    alignItems: 'center',
                }}>
                <p>Error in Data Base</p>
            </Box>
        );
    }

    return (
        <Box
            sx={{
                minHeight: '72vh',
                width: '100%',
                position: 'relative',
                pb: '2rem',
            }}>
            <Box
                sx={{
                    bgcolor: '#C3D600',
                    height: presentationImage !== '#' ? '18rem' : '15rem',
                    width: '100%',
                    position: 'absolute',
                }}></Box>
            <Container sx={{ position: 'relative' }}>
                {title ? (
                    <Typography component='div'>
                        <Box
                            sx={{
                                display: 'flex',
                                justifyContent: 'center',
                                alignItems: 'center',
                                textAlign: 'center',
                                height: presentationImage !== '#' ? '10rem' : '15rem',
                                color: '#fff',
                                fontWeight: 'bold',
                                fontSize: '2rem',
                            }}>
                            {title}
                        </Box>
                    </Typography>
                ) : (
                    <Skeleton variant='rectangular' width={'100%'} height={'15rem'} />
                )}
            </Container>
            <Container
                fixed
                sx={{
                    position: 'relative',
                    py: presentationImage !== '#' ? '0rem' : '2rem',
                }}>
                {presentationImage !== '#' ? (
                    <Box
                        sx={{
                            maxHeight: '20rem',
                            display: 'flex',
                            justifyContent: 'center',
                            pb: '2rem',
                        }}>
                        <img src={presentationImage} alt='Imagen de Presentacion' />
                    </Box>
                ) : null}
                <Box sx={{ display: 'flex', flexDirection: 'row', flexWrap: 'wrap' }}>
                    <Box sx={{ width: matches ? '80%' : '100%' }}>
                        {data_procesed ? (
                            data_procesed.map((element, index) => {
                                return (
                                    <Box
                                        key={index}
                                        sx={{
                                            textAlign: element.includes('src') ? 'center' : 'left',
                                            position: 'relative',
                                            paddingBottom: '1.5rem'
                                        }}>
                                        <Markup content={element} />
                                    </Box>
                                );
                            })
                        ) : (
                            <Skeleton variant='rectangular' width={'100%'} height={'100%'} />
                        )}
                    </Box>
                    <Box sx={{ width: matches ? '20%' : '100%', pl: '2rem' }}>
                        <PriorityDownloads />
                    </Box>
                </Box>
                <div id='news'>
                    <Box textAling='left' sx={styles.titleContainerNews}>
                        <Typography variant='h5'>Ultimas Noticias</Typography>
                    </Box>
                    <Grid container spacing={6} alignItems='stretch'>
                        {dataNews ? (
                            dataNews.news.map((newData) => (
                                <Grid
                                    key={newData.id}
                                    item
                                    xs={12}
                                    sm={6}
                                    md={4}
                                    sx={{
                                        align: 'center',
                                    }}>
                                    <CardNews
                                        descriptionInfo
                                        key={newData.id}
                                        data={newData}
                                        maxWidth={'345'}
                                    />
                                </Grid>
                            ))
                        ) : (
                            <Skeleton variant='rectangular' width={'100%'} height={'100%'} />
                        )}
                    </Grid>
                </div>
            </Container>
        </Box>
    );
};
