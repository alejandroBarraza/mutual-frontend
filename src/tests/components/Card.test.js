import { render, cleanup, waitFor } from '@testing-library/react';
import { MockedProvider } from '@apollo/client/testing';
import '@wojtekmaj/enzyme-adapter-react-17';
import { CardNews } from '../../components/news/CardNews';
import { MemoryRouter } from 'react-router';
import { GETNEWS_START_LIMIT } from '../../Graphql/Queries';
afterEach(cleanup);

const mocks = [
    {
        request: {
            query: GETNEWS_START_LIMIT,
            variables: {
                start: 0,
                limit: 3,
            },
        },
        result: {
            data: {
                news: {
                    id: '3',
                    titulo: 'notica de prueba 1',
                    descripcion: 'la mejor noticia 1',
                    tiempoLectura: 1,
                    imagen: {
                        url: 'https://res.cloudinary.com/dkfgeruxd/image/upload/v1632330012/arquitectura_drawio_735b384d8f.png',
                    },
                },
            },
        },
    },
];

describe('Testing <CardNews/> component ', () => {
    it('render should march with snapshot', async () => {
        const { container } = render(
            <MemoryRouter>
                <MockedProvider mocks={mocks} addTypename={false}>
                    <CardNews data={mocks} descriptionInfo maxWidth={'345'} />
                </MockedProvider>
            </MemoryRouter>
        );
        await waitFor(() => new Promise((res) => setTimeout(res, 0)));
        expect(container).toMatchSnapshot();
    });
});
