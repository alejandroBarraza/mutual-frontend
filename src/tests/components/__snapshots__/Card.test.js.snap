// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing <CardNews/> component  render should march with snapshot 1`] = `
<div>
  <a
    href="/noticias/undefined"
    style="text-decoration: none;"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-3lwjl-MuiPaper-root-MuiCard-root"
    >
      <img
        alt="testing pic"
        class="MuiCardMedia-root MuiCardMedia-media MuiCardMedia-img css-j0zv8p-MuiCardMedia-root"
        height="140"
      />
      <div
        class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
      >
        <div
          class="MuiTypography-root MuiTypography-h5 MuiTypography-gutterBottom css-h93ljk-MuiTypography-root"
        />
        <p
          class="MuiTypography-root MuiTypography-body2 css-r40f8v-MuiTypography-root"
        />
      </div>
      <hr
        class="MuiDivider-root MuiDivider-fullWidth css-ac0cew-MuiDivider-root"
      />
      <div
        class="MuiBox-root css-1bvnc3y"
      >
        <h6
          class="MuiTypography-root MuiTypography-subtitle2 css-ati914-MuiTypography-root"
        >
          undefinedmin
           
        </h6>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeSmall css-ltyecm-MuiSvgIcon-root"
          data-testid="MenuBookRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.5 4.5c-1.95 0-4.05.4-5.5 1.5-1.45-1.1-3.55-1.5-5.5-1.5-1.45 0-2.99.22-4.28.79C1.49 5.62 1 6.33 1 7.14v11.28c0 1.3 1.22 2.26 2.48 1.94.98-.25 2.02-.36 3.02-.36 1.56 0 3.22.26 4.56.92.6.3 1.28.3 1.87 0 1.34-.67 3-.92 4.56-.92 1 0 2.04.11 3.02.36 1.26.33 2.48-.63 2.48-1.94V7.14c0-.81-.49-1.52-1.22-1.85-1.28-.57-2.82-.79-4.27-.79zM21 17.23c0 .63-.58 1.09-1.2.98-.75-.14-1.53-.2-2.3-.2-1.7 0-4.15.65-5.5 1.5V8c1.35-.85 3.8-1.5 5.5-1.5.92 0 1.83.09 2.7.28.46.1.8.51.8.98v9.47z"
          />
          <path
            d="M13.98 11.01c-.32 0-.61-.2-.71-.52-.13-.39.09-.82.48-.94 1.54-.5 3.53-.66 5.36-.45.41.05.71.42.66.83-.05.41-.42.71-.83.66-1.62-.19-3.39-.04-4.73.39-.08.01-.16.03-.23.03zm0 2.66c-.32 0-.61-.2-.71-.52-.13-.39.09-.82.48-.94 1.53-.5 3.53-.66 5.36-.45.41.05.71.42.66.83-.05.41-.42.71-.83.66-1.62-.19-3.39-.04-4.73.39-.08.02-.16.03-.23.03zm0 2.66c-.32 0-.61-.2-.71-.52-.13-.39.09-.82.48-.94 1.53-.5 3.53-.66 5.36-.45.41.05.71.42.66.83-.05.41-.42.7-.83.66-1.62-.19-3.39-.04-4.73.39-.08.02-.16.03-.23.03z"
          />
        </svg>
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeSmall css-p42h9j-MuiSvgIcon-root"
          data-testid="IosShareRoundedIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M18 8h-2c-.55 0-1 .45-1 1s.45 1 1 1h2v11H6V10h2c.55 0 1-.45 1-1s-.45-1-1-1H6c-1.1 0-2 .9-2 2v11c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V10c0-1.1-.9-2-2-2z"
          />
          <path
            d="M12 16c.55 0 1-.45 1-1V5h1.79c.45 0 .67-.54.35-.85l-2.79-2.79c-.2-.2-.51-.2-.71 0L8.85 4.15c-.31.31-.09.85.36.85H11v10c0 .55.45 1 1 1z"
          />
        </svg>
      </div>
    </div>
  </a>
</div>
`;
